tests/audit_for_gaps_test.py

```diff
@@ -0,0 +1,120 @@
+import pytest
+
+from rdm.audit_for_gaps import _find_keys_in_sources, \
+    _read_raw_checklists, _split_out_include_files, _extract_keys_from_checklist, _find_failing_checklist_items
+
+
+@pytest.fixture
+def example_short_checklist_source():
+    return [
+        ('include other_file\napple tempted Eve\nbanana tempted Curious George\n# commentary', 'yellow brick road')
+    ]
+
+
+@pytest.fixture
+def example_long_checklist_source():
+    return [
+        (
+            'include other_file\napple tempted Eve\nbanana tempted Curious George\n# commentary\ncherry\ndates',
+            'yellow brick road'
+        )
+    ]
+
+
+@pytest.fixture
+def example_short_checklist():
+    return [
+        {
+            'reference': 'apple',
+            'description': 'tempted Eve',
+        },
+        {
+            'reference': 'banana',
+            'description': 'tempted Curious George',
+        },
+    ]
+
+
+@pytest.fixture
+def example_long_checklist():
+    return [
+        {
+            'reference': 'apple',
+            'description': 'tempted Eve',
+        },
+        {
+            'reference': 'banana',
+            'description': 'tempted Curious George',
+        },
+        {
+            'reference': 'cherry',
+        },
+        {
+            'reference': 'dates',
+        },
+    ]
+
+
+@pytest.fixture
+def example_raw_checklist():
+    return [
+        {
+            'include': 'other_file',
+            'path': 'yellow brick road'
+        },
+        {
+            'reference': 'apple',
+            'description': 'tempted Eve',
+        },
+        {
+            'reference': 'banana',
+            'description': 'tempted Curious George',
+        },
+    ]
+
+
+document_a = "We like apple pie."
+document_b = "We hate banana splits."
+document_ac = "We like apple pie and cherry pie."
+document_ad = "Never put dates in apple pie."
```

Comment:
How do we handle lines with just whitespace characters?

Also, how do we handle whitespace followed by a comment?

Also, how does a git deal with these situations in .gitignore files? (I suspect their approach will be good)

Reply:
We do not parse the documents, just the checklists.

Reply:
I left my comment in a very weird spot! Sorry, so all of these questions apply to the checklists, not the documents as is implied by where the comment is located.

Reply:
In a checklist any line starting with white space is ignored. We could strip leading white space before parsing. Is that a good idea?

Reply:
I think the way you have it set up is great!